include/neighbor/Runtime.h

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2018-2019, Michael P. Howard.
+// This file is released under the Modified BSD License.
+
+// Maintainer: mphoward
+
+#ifndef NEIGHBOR_RUNTIME_H_
+#define NEIGHBOR_RUNTIME_H_
+
+#include <iostream>
+#include <functional>
+#include <utility>
+
+#include <cuda_runtime.h>
+#ifdef __CUDACC__
+#define GPUCC
+#endif
+
+namespace neighbor
+{
+namespace gpu
+{
+
+/* errors */
+typedef cudaError_t error_t;
+enum error
+    {
+    success=cudaSuccess
+    };
+//! Coerce comparison of gpu::error with the native error as int
+inline bool operator==(error a, error_t b)
+    {
+    return (static_cast<int>(a) == static_cast<int>(b));
+    }
+//! Coerce comparison of gpu::error with the native error as int
+inline bool operator==(error_t a, error b)
+    {
+    return (static_cast<int>(a) == static_cast<int>(b));
+    }
+
+/* streams */
+typedef cudaStream_t stream_t;
+//! Create a GPU stream
+inline error_t streamCreate(stream_t* stream)
+    {
+    return cudaStreamCreate(stream);
+    }
+//! Synchronize a GPU stream
+inline error_t streamSynchronize(stream_t stream)
+    {
+    return cudaStreamSynchronize(stream);
+    }
+//! Destroy a GPU stream
+inline error_t streamDestroy(stream_t stream)
+    {
+    return cudaStreamDestroy(stream);
+    }
+
+/* memory */
+static const int memAttachGlobal = cudaMemAttachGlobal;
+static const int memAttachHost = cudaMemAttachHost;
+//! Allocate GPU memory
+inline error_t malloc(void** ptr, size_t size)
+    {
+    return cudaMalloc(ptr, size);
+    }
+//! Allocate managed GPU memory
+inline error_t mallocManaged(void** ptr, size_t size, unsigned int flags = memAttachGlobal)
+    {
+    return cudaMallocManaged(ptr, size, flags);
+    }
+//! Free GPU memory
+inline error_t free(void* ptr)
+    {
+    return cudaFree(ptr);
+    }
+//! Set GPU memory to a value
+inline error_t memset(void* ptr, int value, size_t count)
+    {
+    return cudaMemset(ptr, value, count);
+    }
+//! Asynchronously set GPU memory to a value
+inline error_t memsetAsync(void* ptr, int value, size_t count, stream_t stream=0)
+    {
+    return cudaMemsetAsync(ptr, value, count, stream);
+    }
+
+/* kernels */
+typedef cudaFuncAttributes funcAttributes;
```

Comment:
We could call this `funcAttributes_t` for consistency

Reply:
I was choosing names by lopping off the `cuda` or `hip` prefix since I thought that might make it easier to convert code later, hence `cudaFuncAttributes -> funcAttributes`. I'm also OK with `funcAttributes_t` though. Which do you think is easier to understand long term?